setup challenge

gmanchon opened the pull request:
adds setup challenge with working tests

move shell challenge to unit 2
move git challenge to unit 3

remove mlflow viz challenge

@krokrob I would like to add some tests that only run locally. do you know if there is an env var that we could use to detect glovebox ? I thought about docker env variables but we might have students running VMs or containers in the future

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,56 @@
+
+Run the first few tests to verify that:
+- the `gcloud` CLI tool has access to your GCP account
+- python code running on your machine has access to your GCP account
```
```suggestion
- Python code running on your machine has access to your GCP account
```

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,56 @@
+
+The second CLI tool that you will use often allows to handle files stored in the cloud.
+
+Let's create a **bucket** that you will use to store files in the cloud if you do not already have one.
```
Is it a challenge? If so it has to be clearer, what is expected.

I see they should use `gsutil mb gs://BUCKET_NAME`

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,56 @@
+
+The `gcloud` CLI allows you to interact with almost all the ressources of your account (there are other dedicated commands for specific products).
+
+Find the `gcloud` command allowing you to list your **GCP project id**.
```
This is a key takeaway of the challenge so it should appear as such.

E.g.

### ❓How do you list your GCP projects?

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,56 @@
+
+Find the `gcloud` command allowing you to list your **GCP project id**.
+
+Use it to fill the `PROJECT_ID` variable in the `Makefile`.
```
This is an assessment and must be visible

E.g.

### 📝 Fill the `PROJECT_ID` variable in the `Makefile`

krokrob replied:
Then run the test with `make dev_test`

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,56 @@
+
+You can use the [Cloud Storage console](https://console.cloud.google.com/storage/) or use the `gsutil` command allowing you to do so.
+
+Find the `gsutil` command allowing you to retrieve the name of your **bucket**.
```
Key takeaway

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,56 @@
+
+Find the `gsutil` command allowing you to retrieve the name of your **bucket**.
+
+Use it to fill the `BUCKET_NAME` variable in the `Makefile`.
```
Assessment

krokrob replied:
then run `make dev_test`

krokrob reviewed (changes requested):
Good start @gmanchon ! I think the instructions, takeaways and assessment should be more visible in the READMEs. Continue the good work!

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+
+Let's make sure that your setup is operational.
+
+📝 **Run the first `4` tests** with `make dev_test` to verify that:
```
```bash
📝 **Run the first `4` tests** with `make dev_test` in your terminal to verify that:
```

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+📝 **Run the first `4` tests** with `make dev_test` to verify that:
+- The `gcloud` CLI tool has access to your GCP account
+- The python code running on your machine has access to your GCP account
+
```
Do you mean the first 4 tests should be green at this point? If so, let's precise it:
```suggestion

The first 4 tests should be ✅. If they are not, check the _hint_. 

```

And are we sure they will be displayed in the right order?

gmanchon replied:
concerning the order that would be true only because I am not currently running the tests of the package itself...

@krokrob should we have them manually `TEST_ENV=development pytest -s tests/test_setup.py` for this one rather than use the make command ? (or should we create a dedicated make command for this one ?

gmanchon replied:
I'll create a dedicated command :)

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+
+📝 **Fill the `PROJECT_ID` variable in the `Makefile`**
+
+Then run the test with `make dev_test`.
```
We could specify which tests should pass after that point

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+
+## Cloud Storage and the `gsutil` CLI
+
+The second CLI tool that you will use often allows to deal with files stored in the cloud.
```
```suggestion
The second CLI tool that you will use often allows you to deal with files stored in the cloud.
```

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+
+Find the `gsutil` command allowing you to create a **bucket**.
+
+📝 **Create a bucket in your GCP account**
```
```suggestion
💻 **Create a bucket in your GCP account**
```

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+
+📝 **Fill the `BUCKET_NAME` variable in the `Makefile`**
+
+Then run the test with `make dev_test`.
```
All tests should pass! 👏

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -0,0 +1,85 @@
+
+📝 **Create a bucket in your GCP account**
+
+Then run the test with `make dev_test`.
```
Specify which tests should pass or suggest that more tests should pass (8/12 for instance...)

krokrob reviewed (changes requested):
Nice! I have some more suggestion then I'm done, everything will be fine 🙈

gmanchon commented:
@krokrob thanks for your feedback ! 🙏 I renamed the tests to try to have shorter and clearer names, and added a docstring for the TAs, but the errors should be self explanatory. WDYT ?

krokrob commented on versioned/02/02_README_setup.md:
```diff
@@ -75,7 +75,9 @@ Find the `gsutil` command allowing you to retrieve the name of your **bucket**.
 📝 **Fill the `BUCKET_NAME` variable in the `Makefile`**
 
-Then run the test with `make dev_test`.
+Then run the test with `make dev_test` (`test_setup_bucket_name` should be ✅).
```
```suggestion
Then run the tests with `make dev_test` (`test_setup_bucket_name` should be ✅).
```

krokrob commented on model/Makefile:
```diff
@@ -1,3 +1,24 @@
+	PYTHONDONTWRITEBYTECODE=1 pytest -v --color=yes
+
+dev_test:
+	TEST_ENV=development PYTHONDONTWRITEBYTECODE=1 pytest -v --color=yes
```
```
write_results && TEST_ENV=development PYTHONDONTWRITEBYTECODE=1 pytest -v --color=yes
```

gmanchon replied:
I added the [instruction](https://github.com/lewagon/taxi-fare/pull/11/commits/5b42fdc1deb74f98e5196d3bca8e9dfc8bef66c8)

krokrob reviewed (changes requested):
I think we forgot to write the `.txt` file before running the tests...

gmanchon commented:
@krokrob all good 😁
